### PR TITLE
feat: #131 Add check who can edit info

### PR DESCRIPTION
**Pull request by Ekaterina1994**

### CODE REVIEW

**Review by dayanych — APPROVED** [review]

File: mw-webapp/src/logic/wayPage/WayPage.tsx [hunk 1]
```diff
@@ -37,6 +38,8 @@ interface WayPageProps {
 export const WayPage = (props: WayPageProps) => {
   const navigate = useNavigate();
   const [way, setWay] = useState<WayPreview>();
+  const {user} = useUserContext();
+  const isUserCanEditWay = user?.uid === way?.owner.uuid;
```
Comment (Howl404) [inline mw-webapp/src/logic/wayPage/WayPage.tsx:42]: Why different variable name?

**Review by Howl404 — CHANGES_REQUESTED** [review]

File: mw-webapp/src/component/editableText/EditableText.tsx [hunk 2]
```diff
@@ -24,6 +24,11 @@ interface EditableTextProps<T> {
    * Additional custom class name for the editable input
    */
   className?: string;
+
+  /**
+   * Is text editable or not
```
Comment (tritonJS826) [inline mw-webapp/src/component/editableText/EditableText.tsx:29]: 1. Default
2.  What does mean editable? Current mode or that doubleclick handler disabled
3.how does "or not" help?

File: mw-webapp/src/component/editableText/EditableText.tsx [hunk 3]
```diff
@@ -33,6 +38,14 @@ export const EditableText = <T extends string | number>(props: EditableTextProps
   const [isEditing, setIsEditing] = useState(false);
   const [text, setText] = useState<T>(props.text);
 
+  if (!props.isEditable) {
```
Comment (tritonJS826) [inline mw-webapp/src/component/editableText/EditableText.tsx:41]: Why if editing is not allowed you should return something different?

**Review by tritonJS826 — CHANGES_REQUESTED** [review]

File: mw-webapp/src/logic/wayPage/WayPage.tsx (hunk 1 above)
Comment (Ekaterina1994) [reply mw-webapp/src/logic/wayPage/WayPage.tsx]: Renamed

File: mw-webapp/src/component/editableText/EditableText.tsx (hunk 2 above)
Comment (Ekaterina1994) [reply mw-webapp/src/component/editableText/EditableText.tsx]: Added default and explained what this property means

File: mw-webapp/src/logic/wayPage/reportsTable/WayColumns.tsx [hunk 4]
```diff
@@ -84,7 +84,7 @@ export const Columns = (props: ColumnsProps) => {
   const ownerName = props.way.owner.name;
   const isOwner = user?.uid === ownerUuid;
   const isMentor = user ? !!props.mentors.get(user.uid) : false;
-  const isUserCanEditComments = (isOwner || isMentor) && user;
+  const isUserCanAddComments = !!user && (isOwner || isMentor);
```
[the remark this answers is not in the document]
Comment (Ekaterina1994) [reply mw-webapp/src/logic/wayPage/reportsTable/WayColumns.tsx:87]: I don't know how to fix it then

File: mw-webapp/src/component/editableText/EditableText.tsx [hunk 5]
```diff
@@ -24,6 +24,12 @@ interface EditableTextProps<T> {
    * Additional custom class name for the editable input
    */
   className?: string;
+
+  /**
+   * If false - doubleclick handler disabled, if true - doubleclick handler allowed
+   * @default false
+   */
+  isEditable?: boolean;
```
[the remark this answers is not in the document]
Comment (Ekaterina1994) [reply mw-webapp/src/component/editableText/EditableText.tsx:32]: Changed

File: mw-webapp/src/logic/wayPage/reportsTable/WayColumns.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (Ekaterina1994) [reply mw-webapp/src/logic/wayPage/reportsTable/WayColumns.tsx]: Refactored

**Review by tritonJS826 — APPROVED** [review]

**Review by Howl404 — APPROVED** [review]

**Pull request merged**